### PR TITLE
parse `v"∞"` as `typemax(VersionNumber)`

### LINKED ISSUE #23027: `v"∞"` doesn't parse
Since we print `typemax(VersionInfo)` this way, seems like we should accept it:
```
julia> typemax(VersionNumber)
v"∞"

julia> v"∞"
ERROR: ArgumentError: invalid version string: ∞
```
alternatively, we could print the former as `typemax(VersionInfo)`.

### PR DESCRIPTION
**Pull request by StefanKarpinski**
Fixes #23027

### CODE REVIEW

**Comment by Keno** [comment]
Fixes #23027

**Comment by martinholters** [comment]
Oh boy, looks like this _really, really_ fixes #23027 😄

**Comment by Keno** [comment]
I think those were three simultaneous edits to add that in various places.

File: test/version.jl [hunk 1]
```diff
@@ -94,8 +94,7 @@ show(io,v"4.3.2+1.a")
 
 # typemin and typemax
 @test typemin(VersionNumber) == v"0-"
-@test typemax(VersionNumber) ==
-  VersionNumber(typemax(Int), typemax(Int), typemax(Int), (), ("",))
```
Comment (tkelman) [inline test/version.jl]: this should probably keep this line

Comment (StefanKarpinski) [inline test/version.jl]: Added back in https://github.com/JuliaLang/julia/pull/23048.

Comment (tkelman) [inline test/version.jl]: Sure, but that's not directly tested either from what I can tell. In the unlikely event that some bug or mistake happened to make `typemax(VersionNumber)` start outputting, say, `v"5.0.0"`, would be good to see an obvious test failure from it.

**Pull request merged**